internal/server/GameLoop.go

```diff
@@ -3,30 +3,100 @@ package server
 import (
 	"SOMAS2023/internal/common/objects"
 	"SOMAS2023/internal/common/physics"
+	"SOMAS2023/internal/common/utils"
 	"fmt"
+
+	"github.com/google/uuid"
 )
 
 func (s *Server) RunGameLoop() {
-	// Each agent makes a decision
+
+	// -------------------------- PROCESS JOINING REQUESTS -------------------------
+	// 1. group agents that have onBike = false by the bike they are trying to join
+	bikeRequests := s.GetJoiningRequests()
+	// 2. pass to agents on each of the desired bikes a list of all agents trying to join
+	for bike, pendingAgents := range bikeRequests {
+		agents := s.megaBikes[bike].GetAgents()
+
+		responses := make([](map[uuid.UUID]bool), 0) // list containing all the agents' ranking
+		for _, agent := range agents {
+			responses = append(responses, agent.DecideJoining(pendingAgents))
+		}
+		// 3. accept agents based on the response outcome (it will have to be a ranking system, as only 8-n bikers can be accepted)
+		acceptedRanked := utils.GetAcceptanceRanking(responses)
+		for _, accepted := range acceptedRanked[:(8 - len(s.megaBikes[bike].GetAgents()))] {
```

Comment:
Split into variables for readability

Reply:
`totalSeatsFilled := len(s.megaBikes[bike].GetAgents()) `
`spareSeats := 8 - totalSeatsFilled`